src/polymer-ide.ts

```diff
@@ -61,21 +68,142 @@ class PolymerIde {
+  documentationElement: HTMLElement = null;
+  editorService: RemoteEditorService;
+  editorElements: Array<HTMLElement> = [];
+
+  constructor(editorService: RemoteEditorService) {
+    this.editorService = editorService;
+  };
+
+  subscribe(subscriptions: CompositeDisposable) {
+    subscriptions.add(atom.workspace.observePanes((pane: AtomCore.IPane) => {
+      this.update();
+      subscriptions.add(pane.onDidChangeActiveItem(() => {
+        this.update();
+      }));
+    }));
+  };
+
+  update() {
+    this.removeTooltip();
+    this.textEditor = atom.workspace.getActiveTextEditor();
+    this.addMouseEventListener();
+  }
+
+  addMouseEventListener() {
+    const textEditorElement = atom.views.getView(this.textEditor);
+    if (!textEditorElement) {
+      return;
+    }
+    if (this.editorElements.indexOf(textEditorElement) > -1) {
+      return;
+    }
+    this.editorElements.push(textEditorElement);
+    let timer: number;
+    textEditorElement.addEventListener('mousemove', (event: AtomMouseEvent) => {
+      window.clearTimeout(timer);
+      this.removeTooltip();
+      timer = window.setTimeout(() => this.calculatePositions(event, textEditorElement), 100);
+    });
+  };
+
+  calculatePositions(event: AtomMouseEvent, textEditorElement: any) {
+    // The text editor could have been closed before the timeout was fired
+    if (!textEditorElement || !textEditorElement.component) {
+      return;
+    }
+    const reportedCursorPosition = textEditorElement.component.screenPositionForMouseEvent(event);
+    const currentCursorPixelPosition = textEditorElement.component.pixelPositionForMouseEvent(event);
+    const expectedPixelPosition = textEditorElement.pixelPositionForScreenPosition(reportedCursorPosition);
+
+    const nearColumn = Math.abs(currentCursorPixelPosition.left - expectedPixelPosition.left) < 20;
+    const nearRow = Math.abs(currentCursorPixelPosition.top - expectedPixelPosition.top) < 20;
+
+    if (nearColumn && nearRow) {
+      this.updateTooltip(this.textEditor.bufferPositionForScreenPosition(reportedCursorPosition), event);
+    }
+  }
+
+  async updateTooltip(point: {row: number, column: number}, event: AtomMouseEvent) {
+    this.removeTooltip();
+
+    if (!(this.editorService && this.textEditor)) {
+      return;
+    }
+    const relativePath: string =
+      atom.project['relativizePath'](this.textEditor.getPath())[1];
+    let documentation: string;
+    try {
+      documentation = await this.editorService.getDocumentationAtPosition(relativePath, {line: point.row, column: point.column});
+      if (!documentation) {
+        return;
+      }
+    } catch (e) {
+      // The analyzer was not able to parse the file
+      return;
+    }
+
+    const marker = this.textEditor.markBufferRange([point, point]);
+    const div = this.getOrCreateTooltipElement();
+    div.style.left = event.left;
+    div.style.top = event.top;
+    this.documentationElement.innerHTML = marked.parse(documentation);
```

Comment:
Need to set `strict` to true here in the call to `marked`, both to prevent unintended rendering, and so that a clever library author can't hack your editor